backupServer.py

```diff
@@ -0,0 +1,128 @@
+import platform
+import os 
+import datetime
+import shutil
+
+
+date = datetime.datetime.now()
+system=  platform.system()
+
+
+
+BackupData= ''
+
+BackupVorm = input('hoe wilt u uw data backuppen opslaan 1) Local 2) Op de backupserver : ')
+
+
+def BackupDirectory(sftp):
+
+        DirPath = input('\ngeef hier de volledige pad waar u een backup van wilt opslaan: ')
+        BackupDestination = input('\ngeef hier vervolgens de bestemmings locatie/pad van de directory waar u de backup wilt hebben (sluit af met /): ')
+
+        DirFiles = os.listdir(DirPath)
+        os.chdir(DirPath) #Veranderd huidige positie naar de backup directory
+        # ga elke file af in de opgevraagde director/ 1 voor 1 kopieren
+        for file in DirFiles:
+                CheckIfDir = os.path.isdir(file) # controleerd of file een directory is
+
+                # Als file geen directory is kopier naar backup locatie
+                if CheckIfDir == False:
+
+                        # check of de backup local of op de remote server plaats vind
+                        if BackupVorm == '1':
+                                shutil.copy(file , BackupDestination)
+                        elif BackupVorm == '2': #als server gebruik put functie in sftp
+                                sftp.put(file,os.path.join(BackupDestination, file)) 
+
+                else: # als file een directory is doe niks/ sla over
```

Comment:
Het zou beter zijn om iets te maken zodat de dir alsnog gedaan word.
Mischien door eerst een nieuwe dir aan te maken met de zelfde naam en dan de info daarin te zetten

Reply:
De rede dat hij niks doet met een dir is omdat dit niet nodig is, want hij kopiert alle files in een dir. Maar het is wel een goed idee indd om een directory op de backuplocatie aan te maken met een zelfde naam. Hier ga ik morgen naar kijken